tokio/src/task/join_set.rs

```diff
@@ -306,6 +306,22 @@ impl<T: 'static> JoinSet<T> {
         crate::future::poll_fn(|cx| self.poll_join_next_with_id(cx)).await
     }
 
+    /// Tries to join one of the tasks in the set that has completed and return its output.
+    ///
+    /// Returns `None` if the set is empty.    
+    pub fn try_join_next(&mut self) -> Option<Result<T, JoinError>> {
+        let mut entry = self.inner.try_pop_notified()?;
+
+        let res = entry.with_value_and_context(|jh, ctx| Pin::new(jh).poll(ctx));
+
+        if let Poll::Ready(res) = res {
+            let _entry = entry.remove();
+            Some(res)
+        } else {
+            None
+        }
```

Comment:
This case happens if we successfully pop a `JoinHandle` from the list of notified tasks, but it turns out that the `JoinHandle` is not actually ready yet. In this case, we should not return `None` because there might be another `JoinHandle` in the list of notified task that is actually ready.

Instead, please wrap this function in a loop and go around the loop to try again in this case.

Reply:
You're right. Made the adjustments.